### PR TITLE
identity: add volume replication capability

### PR DESCRIPTION
**Pull request by yati1998**
This commit adds volume replication capability to the
identity service as the part of moving the operator
to kuberenetes-csi-addons repo and being used as a service
similar to other services like reclaim space and network fnece.

Fixes: #41

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

File: identity/identity.proto [hunk 1]
```diff
@@ -139,15 +139,38 @@ message Capability {
     Type type = 1;
   }
 
+  // VolumeReplication contains the features of the volumereplication operation that the
```
Comment (nixpanic) [inline identity/identity.proto:142]: CI fails:
```
[identity/identity.proto:142:1] The line length is 88, but it must be shorter than 80
[identity/identity.proto:147:1] The line length is 85, but it must be shorter than 80
```

File: identity/identity.proto [hunk 2]
```diff
@@ -139,15 +139,38 @@ message Capability {
...
+      // operation in the current mode. The CSI-Addons CO plugin will most
+      // likely ignore this node for the VolumeReplication operation.
+      UNKNOWN = 0;
+
+      // VOLUME_REPLICATION indicates that the CSI-driver provides RPCs for a
+      // VolumeReplication operation.
+      // The presence of this capability determines whether the CSI-Addons CO
+      // plugin can invoke RPCs that require access to the storage system,
+      // similar to the CSI Controller (provisioner).
+      VOLUME_REPLICATION = 1;
```
Comment (nixpanic) [inline identity/identity.proto:157]: maybe replication has optional features, like:
- promote/demote: some storage providers might not need it
- resync volume: could be optional too?

Comment (yati1998) [reply identity/identity.proto:157]: For now we can go with this to keep it simple and once the migration is done we can decide on adding these options.

Comment (nixpanic) [reply identity/identity.proto:157]: ok, but note that making something optional after the fact is not as easy as doing it from the start

**Review by nixpanic — APPROVED** [review]
Please get @ShyamsundarR and/or @Madhu-1 to approve this too. Thanks!

**Comment by yati1998** [comment]
@Rakshith-R

File: identity/identity.proto (hunk 2 above)
Comment (Madhu-1) [reply identity/identity.proto:158]: This works for new now to advertise that the csi driver supports Volume_replication. 

We will have two options to discover optional supports 

* we can introduce one more RPC call to check what different capabilities if the volume Replication is supported, the driver can respond what different caps (enable/disable, promote/demote,resync) csi driver supports

* And more fields to the VolumeReplication for individual caps.

File: identity/README.md [hunk 3]
```diff
@@ -239,6 +239,28 @@ message Capability {
...
   }
 
+  // VolumeReplication contains the features of the volumereplication operation
+  // that the CSI-driver supports.
+  message VolumeReplication {
+    // Type describes a CSI Service that CSI-drivers can support.
+    enum Type {
+      // UNKNOWN indicates that the CSI-driver does not support the
+      // VolumeReplication operation in the current mode. The CSI-Addons CO
+      // plugin will most likely ignore this node for the
```
Comment (Madhu-1) [inline identity/README.md:249]: ` this node`? need to remove it?

Comment (yati1998) [reply identity/README.md:249]: oh yeah, will remove.

Comment (Rakshith-R) [reply identity/README.md:249]: The phrase is similar to reclaimspace and network fence ? 

This LGTM

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Comment by yati1998** [comment]
@nixpanic can you please review the PR.

**Review by nixpanic — APPROVED** [review]

**Pull request merged**